python_legacy/iceberg/core/base_table_scan.py

```diff
@@ -125,6 +125,26 @@ def option(self, property, value):
                                      selected_columns=self.selected_columns, options=builder,
                                      minused_cols=self.minused_cols)
 
+    def check_partition_exist(self, ops=None, snapshot=None, row_filter=None):
+
+        if not all(i is None for i in [ops, snapshot, row_filter]):
+            raise NotImplementedError()
```

Comment:
Why is this a not implemented error, looks more like `IllegalArgumentException`.